### ISSUE TITLE
azurerm_api_management_api ~ "ApiVersionSet not found" on recreate

### ISSUE THREAD

**Issue by chrisbloe** [hashicorp/terraform-provider-azurerm#21498]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.4.2

### AzureRM Provider Version

3.49.0

### Affected Resource(s)/Data Source(s)

- `azurerm_api_management_api`
- `azurerm_api_management_api_version_set`

### Terraform Configuration Files

```hcl
resource "azurerm_api_management_api" "api" {
  for_each              = toset(local.api_versions)  
  name                  = lower("${var.api_name}-${each.key}")
  resource_group_name   = data.azurerm_resource_group.rg.name
  api_management_name   = data.azurerm_api_management.apim.name
  revision              = replace(trimspace(var.revision), ".", "-")
  display_name          = lower("${var.api_name}-${each.key}")
  path                  = lower("${var.api_name}")
  protocols             = ["https"]
  subscription_required = var.require_sub_key
  service_url           = lower("${var.agw_url}")
  version               = lower("${each.key}")
  version_set_id        = replace(azurerm_api_management_api_version_set.vset.id, "api-version-sets", "apiVersionSets")  

  import {
    content_format  = "openapi+json"
    content_value   = data.http.swagger[each.key].response_body
  }
}
```

### Debug Output/Panic Output

```shell
Can't obtain
```

### Expected Behaviour

When a revision changes I would expect to see a successful replacement after the first attempt:
```
Terraform will perform the following actions:

  # module.api-management-api.azurerm_api_management_api.api["v1"] must be replaced
-/+ resource "azurerm_api_management_api" "api" {
      ~ revision              = "xxx" -> "yyy" # forces replacement
        ...
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

### Actual Behaviour

```
module.api-management-api.azurerm_api_management_api.api["v1"]: Still creating... [4m10s elapsed]
╷
│ Error: waiting on creating/updating Api: (Name "xxx" / Service Name "xxx" / Resource Group "xxx"): Future#WaitForCompletion: the number of retries has been exceeded: StatusCode=404 -- Original Error: Code="ResourceNotFound" Message="ApiVersionSet not found."
│ 
│   with module.api-management-api.azurerm_api_management_api.api["v1"],
│   on ../../modules/apim-integration/main.tf line 68, in resource "azurerm_api_management_api" "api":
│   68: resource "azurerm_api_management_api" "api" {
│ 
╵
##[error]Error: The process '/opt/hostedtoolcache/terraform/1.4.2/x64/terraform' failed with exit code 1
Finishing: terraform apply
```

On the 2nd attempt, the following is seen:

```
**********************************************************************************************************
Terraform detected the following changes made outside of Terraform since the last "terraform apply" which may have affected this plan:
module.api-management-api.azurerm_api_management_api_version_set.vset has been deleted
....
module.api-management-api.azurerm_api_management_api_version_set.vset: Creating...
**********************************************************************************************************
```

This implies that in the first attempt, the version set gets deleted, however it's not recorded in the tfstate due to which it considers the version set to still be present, so tries to create the api, but fails with "ApiVersionSet not found" because it's actually already been deleted.

I'm also wondering if the following line is missing a character:

https://github.com/hashicorp/terraform-provider-azurerm/blob/c125413c6b16d8f7d219032feb121570ac3d6020/internal/services/apimanagement/migration/api_version_set.go#L65

i.e. should `api-version-set` be changed to `api-version-sets`?

Either way, running the same apply twice is successful on the 2nd run.

### References

- #11093
- https://github.com/hashicorp/terraform-provider-azurerm/issues/6609#issuecomment-629551582

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against an early major version of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
